src/applications/bmqstoragetool/bmqstoragetool.m.cpp

```diff
@@ -0,0 +1,197 @@
+         "lower timestamp bound",
+         balcl::TypeInfo(&arguments.d_timestampGt),
+         balcl::OccurrenceInfo::e_OPTIONAL},
+        {"timestamp-lt",
+         "timestamp less than",
+         "higher timestamp bound",
+         balcl::TypeInfo(&arguments.d_timestampLt),
+         balcl::OccurrenceInfo::e_OPTIONAL},
+        {"outstanding",
+         "only outstanding",
+         "show only outstanding (not deleted) messages",
+         balcl::TypeInfo(&arguments.d_outstanding),
+         balcl::OccurrenceInfo::e_OPTIONAL},
+        {"confirmed",
+         "only confirmed",
+         "show only messages, confirmed by all the appId's",
+         balcl::TypeInfo(&arguments.d_confirmed),
+         balcl::OccurrenceInfo::e_OPTIONAL},
+        {"partially-confirmed",
+         "only partially confirmed",
+         "show only messages, confirmed by some of the appId's",
+         balcl::TypeInfo(&arguments.d_partiallyConfirmed),
+         balcl::OccurrenceInfo::e_OPTIONAL},
+        {"details",
+         "details",
+         "specify if you need message details",
+         balcl::TypeInfo(&arguments.d_details),
+         balcl::OccurrenceInfo::e_OPTIONAL},
+        {"dump-payload",
+         "dump payload",
+         "specify if you need message payload",
+         balcl::TypeInfo(&arguments.d_dumpPayload),
+         balcl::OccurrenceInfo::e_OPTIONAL},
+        {"dump-limit",
+         "dump limit",
+         "limit of payload output",
+         balcl::TypeInfo(&arguments.d_dumpLimit),
+         balcl::OccurrenceInfo(1024)},
+        {"summary",
+         "summary",
+         "summary of all matching messages (number of outstanding messages "
+         "and other statistics)",
+         balcl::TypeInfo(&arguments.d_summary),
+         balcl::OccurrenceInfo::e_OPTIONAL},
+        {"h|help",
+         "help",
+         "print usage)",
+         balcl::TypeInfo(&showHelp),
+         balcl::OccurrenceInfo::e_OPTIONAL}};
+
+    balcl::CommandLine commandLine(specTable);
+    if (commandLine.parse(argc, argv) != 0 || showHelp) {
+        commandLine.printUsage();
+        return false;  // RETURN
+    }
+
+    bsl::string error;
+    if (!arguments.validate(&error)) {
+        bsl::cerr << "Arguments validation failed:\n" << error;
+        return false;  // RETURN
+    }
+
+    return true;
+}
+
+// ====
+// main
+// ====
+
+int main(int argc, const char* argv[])
+{
+    // Init allocator
+    bslma::Allocator* allocator = bslma::Default::allocator();
+
+    // Arguments parsing
+    CommandLineArguments arguments(allocator);
+    if (!parseArgs(arguments, argc, argv)) {
+        return 1;  // RETURN
+    }
+
+    // Create parameters
+    bsl::shared_ptr<Parameters> parameters;
+    try {
+        parameters.reset(new (*allocator) Parameters(arguments, allocator),
+                         allocator);
+    }
+    catch (const bsl::exception& e) {
+        bsl::cerr << e.what();
+        return 2;  // RETURN
+    }
```

Comment:
Do we still need all this machinery? Isn't just `Parameters parameters;` enough?